File: parqueadero-api/src/models/vehiculo.js
export class Vehiculo {
  constructor({ id, placa, tipo, propietarioNombre, propietarioTelefono, createdAt, updatedAt }) {
    this.id = id;
    this.placa = placa;
    this.tipo = tipo;
    this.propietarioNombre = propietarioNombre;
    this.propietarioTelefono = propietarioTelefono;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
  }

  static toDomain(record) {
    return new Vehiculo({
      id: record.id,
      placa: record.placa,
      tipo: record.tipo,
      propietarioNombre: record.propietarioNombre,
      propietarioTelefono: record.propietarioTelefono,
      createdAt: record.createdAt,
      updatedAt: record.updatedAt,
    });
  }

  static toPersistence({ placa, tipo, propietarioNombre, propietarioTelefono } = {}) {
    const data = {};
    if (placa !== undefined) data.placa = placa;
    if (tipo !== undefined) data.tipo = tipo;
    if (propietarioNombre !== undefined) data.propietarioNombre = propietarioNombre;
    if (propietarioTelefono !== undefined) data.propietarioTelefono = propietarioTelefono;
    return data;
  }
}

File: parqueadero_app/lib/features/tickets/domain/vehiculo.dart
import '../../../core/domain/tipo_vehiculo.dart';

class Vehiculo {
  const Vehiculo({
    required this.id,
    required this.placa,
    required this.tipo,
    this.propietarioNombre,
    this.propietarioTelefono,
    required this.createdAt,
    required this.updatedAt,
  });

  final String id;
  final String placa;
  final TipoVehiculo tipo;
  final String? propietarioNombre;
  final String? propietarioTelefono;
  final DateTime createdAt;
  final DateTime updatedAt;

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is Vehiculo &&
          runtimeType == other.runtimeType &&
          id == other.id &&
          placa == other.placa &&
          tipo == other.tipo &&
          propietarioNombre == other.propietarioNombre &&
          propietarioTelefono == other.propietarioTelefono &&
          createdAt == other.createdAt &&
          updatedAt == other.updatedAt;

  @override
  int get hashCode =>
      Object.hash(id, placa, tipo, propietarioNombre, propietarioTelefono, createdAt, updatedAt);
}
